README.md

```diff
@@ -6,61 +6,45 @@ The current set of extensions can be read on [GitHub Pages](https://ocfl.github.
-* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory (if parameters are defined)  
-
-## Referencing Parameters
-
-Wherever a parameterised extension is referenced, any parameters MAY be included in an accompanying sidecar JSON file that
-uses the registered name with the filename extension `.json`. If using an 'extensions' directory, the JSON file MUST be 
-included in the root directory of the extension and not a subdirectory. Another place that an extension may be 
-referenced is `ocfl_layout.json` where the sidecar file should be alongside it in the Storage Root. 
-
-For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
-
-    "0000-example-extension": {  
-        "firstExampleParameter": 12,  
-        "secondExampleParameter": "Hello",  
-        "thirdExampleParameter": "Green"  
-    }
-    
-An extension MAY be referenced without providing any parameters, even if they are defined for that extension. However, if parameters are defined they MUST be complete, with valid values specified for all parameters that do not have a default. There MUST only be one parameter set defined for each extension reference.
+### Headers
+
+Extensions MUST contain a header section that defines the following fields:
+
+* **Extension Name**: The extension's unique *Registered Name*
+* **Authors**: The names of the individuals who authored the extension
+* **Minimum OCFL Version**: The minimum OCFL version that the extension requires, eg. *1.0*
+* **Obsoletes**: The *Registered Name* of the extension that this extension obsoletes, or *n/a*
+* **Obsoleted by**: The *Registered Name* of the extension that obsoletes this extension, or *n/a*
+
+### Parameters
+
+Extensions MAY define parameters to enable configuration as needed. Extension parameters are serialized as JSON values, and therefore must conform to the [JSON specification](https://tools.ietf.org/html/rfc8259). Parameters MUST be defined in the following structure:
+
+* **Name**: A short, descriptive name for the parameter. The name is used as the parameter's key within its JSON representation.
+   * **Description**: A brief description of the function of the parameter. This should be expanded on in the main description of the extension which MUST reference all the parameters.
+   * **Type**: The JSON data type of the parameter value. One of `string`, `number`, `boolean`, `array`, or `object`. The structure of complex types MUST be further described.
+   * **Constraints**: A description of any constraints to apply to parameter values. For example, "May not be empty", "Must be greater than 1 and less than 100", or "Must be one of 'red', 'blue', 'green'." 
```

Comment:
I wonder whether we could reference some existing work to define constraints. The three examples given in text would easy to implement in [JSON Schema](https://www.ietf.org/archive/id/draft-handrews-json-schema-02.txt) for example, although I note that this exists only in draft status. Maybe there are other options?

Reply:
@zimeon I'm not aware of any other options. I don't mind if we want to use JSON Schema. I guess, all I would say is why doesn't the OCFL spec itself when it defines its JSON structures? Yes, there is a JSON Schema that's referenced in a non-normative note, but it's more of an extra. To me, I view this as more or less the same situation, and that perhaps a JSON schema should just be included as an additional file.

Reply:
I'm not sure we need something extra like JSON Schema to define constraints - I'm fine with extensions just describing in words what a parameter value should be.

Reply:
To the degree possible, it would be helpful if the "constraint" were unambiguous. An English text description leaves potentially too much room for debate. In the absence of another structured mechanism for describing the constraints, the draft [JSON Schema](https://www.ietf.org/archive/id/draft-handrews-json-schema-validation-02.txt) seems like a reasonable fit.

Reply:
@awoods @zimeon So, you'd like the parameters section of the example extension to look like this:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Example Extension",
  "type": "object",
  "required": [
    "firstExampleParameter"
  ],
  "properties": {
    "firstExampleParameter": {
      "type": "number",
      "description": "a mandatory 8 bit unsigned value",
      "minimum": 0,
      "maximum": 255
    },
   "secondExampleParameter": {
      "type": "string",
      "description": "an optional 64 character long string, defaulting to \"Not applicable\", if omitted.",
      "default": "Not applicable",
      "maxLength": 64
    },
    "thirdExampleParameter": {
      "type": "string",
      "description": "An example enumerated parameter",
      "default": "Green",
      "enum": [ "Red", "Yellow", "Orange", "Green", "Blue", "Indigo", "Violet" ]
  }
}
```

Is that right?

To me, it seems out of place in a document that it otherwise plain english. And, to reiterate my question again, if using JSON Schema like this is such a great fit, why does the OCFL spec not do so as well? Why isn't [this schema](https://github.com/OCFL/spec/blob/master/1.0/spec/inventory_schema.json) pasted directly into the spec [here](https://ocfl.io/1.0/spec/#inventory)?

Reply:
Your example above, @pwinckles , seems very clear. To your point around the awkwardness of having a schema embedded in an otherwise text document, I can see the utility of describing the parameters in the extension definition, and referencing an accompanying schema document. The same could be true in the OCFL spec.

Reply:
So what is the action? To require that extensions provide `NNNN-RegisteredName-schema.json` in addition to the MD file?

Reply:
I'm ready to withdraw my initial comment. My particular objection was to defining our own constraints language (that this PR removes). If we have an plain text description then that is very general. If an extension writer wanted to use a particular technical description then they could include something like _"Must match regex /^[a-z][a-z0-9+-]*$/"_

Reply:
Suggested language is to remove the sentence that begins `For example...` and update with something like `Constraints may be in plain text, regex, JSON Schema, or whatever makes the most sense for the extension.`